[PATCH] ntfs-3g: force alignment of cached structures for pointers and u64

 Ref: http://tuxera.com/forum/viewtopic.php?f=3&t=7914

---
diff --git a/include/ntfs-3g/cache.h b/include/ntfs-3g/cache.h
index 67c3325..67e4f9d 100644
--- a/include/ntfs-3g/cache.h
+++ b/include/ntfs-3g/cache.h
@@ -29,7 +29,11 @@ struct CACHED_GENERIC {
 	struct CACHED_GENERIC *previous;
 	void *variable;
 	size_t varsize;
-	void *fixed[0];
+	union {
+		/* force alignment for pointers and u64 */
+		u64 u64align;
+		void *ptralign;
+	} fixed[0];
 } ;
 
 struct CACHED_INODE {
-- 
1.6.5.GIT
